src/jit/emitxarch.cpp

```diff
@@ -10832,6 +10832,9 @@ size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
 
             dst += emitOutputByte(dst, emitGetInsSC(id));
             sz = emitSizeOfInsDsc(id);
+
+            // Kill any GC ref in the destination register.
+            emitGCregDeadUpd(id->idReg1(), dst);
```

Comment:
> we can avoid call overhead for those instructions that write to an sse/avx reg by conditioning this on 

Seems reasonable.

> I see emitInsCanOnlyWriteSSE2OrAVXReg() needs to be updated to handle pextrw and pmovkskb instructions.

This is the fix for at least one other piece of #8860. I'm happy to fold it into this change as well.